helix-term/src/main.rs

```diff
@@ -115,25 +115,69 @@ FLAGS:
-                use std::io::Read;
-                let _ = std::io::stdin().read(&mut []);
-                Config::default()
-            }),
-        Err(err) if err.kind() == std::io::ErrorKind::NotFound => Config::default(),
-        Err(err) => return Err(Error::new(err)),
+    // We need the config, to get the editor.security.[load|confirm]_local_config flags
+    let root_config = std::fs::read_to_string(helix_loader::config_file())
+        .map(|config| toml::from_str(&config))?
+        .unwrap_or_default();
+
+    // Now, let's load config based on a PathBuf
+    // We will keep it toml::Value so we can use helix_loader::merge_toml_values
+    // This will respect editor.security settings
+    let load_config = |config_path: std::path::PathBuf| {
+        if std::path::Path::new(&config_path).exists() {
+            let mut confirmed = true;
+            if config_path != helix_loader::config_file() {
+                if root_config.editor.security.load_local_config {
+                    if root_config.editor.security.confirm_local_config {
+                        eprintln!(
+                            "Type yes to continue with loading config: {:#?}",
+                            config_path
+                        );
+                        use std::io::Read;
+                        confirmed = match std::io::stdin().read(&mut [1, 2, 3, 4]) {
+                            Ok(n) => n == "yesE".len(),
+                            Err(_) => false,
+                        }
+                    }
+                } else {
+                    confirmed = false;
+                }
+            }
+            if confirmed {
+                log::debug!("Load config: {:?}", config_path);
+                return std::fs::read(config_path).map(|config| toml::from_slice(&config));
+            }
+        }
+        //TODO: There has to be a better way!?
+        std::fs::read("/not/a/valid/path").map(|config| toml::from_slice(&config))
+    };
+
+    // Load each config file
+    let local_config_values = helix_loader::local_config_dirs()
+        .into_iter()
+        .map(|path| path.join("config.toml"))
+        .chain([helix_loader::config_file()])
+        .filter_map(|file| load_config(file).ok())
+        .collect::<Result<Vec<toml::Value>, _>>();
+
+    // Merge configs and return, or alert user of error and load default
+    let config = match local_config_values {
+        Ok(values) => values
+            .into_iter()
+            .fold(toml::Value::Table(toml::value::Table::default()), |a, b| {
+                helix_loader::merge_toml_values(b, a, 3)
+            })
+            .try_into()?,
+        Err(err) => {
+            eprintln!("Bad configs: {:#?}", err);
+            eprintln!("Press <ENTER> to continue with default config");
+            use std::io::Read;
+            // This waits for an enter press.
+            let _ = std::io::stdin().read(&mut []);
+            Config::default()
+        }
     };
```

Comment:
Please move these into a function, it became quite huge.

Reply:
I have refactored the entire thing. Let me know if I should do another pass at the Rust, but to me it is a lot better now. Thanks again for your time and input!

(I don't know what will generate notifications but I will presume my comment and the "resolve conversation" will suffice).